conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1354,7 +1354,13 @@ def build(self):
     assert "-- MYVAR1 MYVALUE1!!" in client.out
 
 
-def test_no_build_type():
+@pytest.mark.parametrize("conanfile_settings",
+                         ['"os", "compiler", "arch"',
+                          '"os", "compiler", "arch", "build_type"'
+                          ])
+@pytest.mark.parametrize("tc_cache_variables",
+                         [True, False])
```

Comment:
This is testing too different use cases. Even if the code of the test is similar enough to do a ``parameterized`` test, the use cases can be quite different. 
The use case I defined in my PR was: "The conanfile doesn't define a ``build_type`` setting, and needs to define a hardcoded one so the build creates that configuration.

There would be other use cases, like why defining ``build_type`` in the ``settings``, and then ignoring it, overriding it in the recipe? This seems more an ill-formed case, creating 2 different package_ids with the same binaries.

It would be much better to provide 4 different tests, each test representing a different use case, and what is the user story that such test is represeting.